### ISSUE TITLE
sso: Need to enable mod auth_pubtkt

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#890]
When testing plinth v0.15.1, I saw that installed web apps were not available at the expected locations. I tried reloading the apache conf, and saw the following error:
```
Jul 02 14:46:34 freedombox systemd[1]: Reloading The Apache HTTP Server.
Jul 02 14:46:34 freedombox apachectl[11604]: AH00526: Syntax error on line 1 of /etc/apache2/includes/freedombox-single-sign-on.conf:
Jul 02 14:46:34 freedombox apachectl[11604]: Invalid command 'TKTAuthPublicKey', perhaps misspelled or defined by a module not included in the server configuration
```
I think we need to enable this apache module during sso module setup.

**Comment by JosephKiranBabu** [comment]
@jvalleroy Here are the steps I followed to reproduce this bug.

- Switched to the 0.15.1 tag. 
- Deleted the entry for sso from plinth_module in the sqlite db and uninstalled libapache2-mod-auth-pubtkt. 
- Then I started plinth. It installed the package libapache2-mod-auth-pubtkt and also created keys.

Maybe you had libapache2-mod-auth-pubtkt installed on your FreedomBox and keys not created because there was already an entry for sso in the plinth_module table. Can you check this?

I took the latest deb file from CI and got the same result.

**Comment by jvalleroy** [comment]
@JosephKiranBabu I built a plinth-0.15.1+ds-1 package (my packaging work-in-progress is at https://github.com/jvalleroy/plinth-debian). Then I built a virtualbox-amd64 image using freedom-maker, with this custom package.

The keys are there in /etc/apache2/auth-pubtkt-keys. The issue is that the auth_pubtkt module is not enabled (it's not linked in /etc/apache2/mods-enabled/).

**Comment by jvalleroy** [comment]
@JosephKiranBabu I think this issue still exists. When I build a testing image with recent plinth, I got an error when trying to access /syncthing. I checked in /etc/apache2/mods-enabled, and saw that auth_pubtkt wasn't listed there. After running ```a2enconf auth_pubtkt``` and restarting apache2, I was able to access the syncthing page.

**Comment by JosephKiranBabu** [comment]
@jvalleroy Looks like the Debian package is enabling the mod, but restarting the Apache server is still required. Please review #1000

**Comment by SunilMohanAdapa** [comment]
Merged #1000 which does enabling of mod_auth_pubtkt during sso module setup for safety.  That should fix the issue. However, we perhaps have to test this again before closing.

**Comment by JosephKiranBabu** [comment]
@jvalleroy @SunilMohanAdapa I've tested this with an `unstable` Raspberry Pi 2 image. SSO is indeed enabled, but I noticed the following 2 issues:
- Reboot during setup is not happening automatically.
- SSO prevents first boot by asking for login before the user can setup their admin account.

These might be unrelated to the above patch or regressions from some other changes. So, I'd like to close this issue and create 2 new issues to track them.
